### ISSUE TITLE
`LinearAttention` fails to give correct results when matrix length=1

### ISSUE THREAD

**Issue by r3dir3ct** [allenai/allennlp#3599]
**Describe the bug**
In `linear_attention._forward_internal`, this line:
`return self._activation(combined_tensors.squeeze(1) + self._bias)`.
It seems `squeeze(1)` here is incorrect.
Because if the `matrix` has the shape `(batch_size, length, encoding_dim)`, the `combined_tensors` will have the shape of `(batch_size, length)`.
If the length>1, the `squeeze` operation is redundant, while the length is 1, the dim of length will be squeezed, which causes the error.
So I don't figure out why there is a strange `squeeze` operation.

**To Reproduce**
```
from allennlp.modules.attention import LinearAttention
attention=LinearAttention(tensor_1_dim=2,tensor_2_dim=2)
vector=torch.Tensor(2,2)
matrix=torch.Tensor(2,1,2)
# shape: (2,1)
mask=torch.tensor([[1],[1]])
# Expected shape (2, 1), but got (2,)
weights=attention(vector,matrix,mask)
```
It's OK to use AdditiveAttention

**System (please complete the following information):**
I installed from master

**Comment by schmmd** [comment]
We would welcome a PR removing that squeeze, as it's unexpected that passing a value of 1 for the dimension changes the number of dimensions in the return. Bear in mind that this might break many of the tests in the library so it might be a little difficult to fix.

**Comment by dirkgr** [comment]
Looks like this was fixed
